src/components/Tool/FixSendNewsletter/Contact.tsx

```diff
@@ -122,15 +125,30 @@ const Contact = ({
         }
       }
     }
-    setNewsletter(newNewsletterValue);
+    updateNewsletterValue(newNewsletterValue);
   }, [primaryAddress]);
 
+  const updateNewsletterValue = (sendNewsletter: SendNewsletterEnum): void => {
+    setNewsletter(sendNewsletter);
+    const existingItem = contactUpdates.find(
+      (contactData) => contactData.id === id,
+    );
+    if (existingItem) {
+      existingItem.sendNewsletter = sendNewsletter;
+    } else {
+      contactUpdates.push({
+        id,
+        sendNewsletter: sendNewsletter,
+      });
+    }
```

Comment:
We shouldn't set state like this as this goes against React's rules. We should use the setContactUpdates to update it.

Using a push mutates the orginial object, it's always best to create a copy and edit the copy and save that as the new version. This can be done with the setContactUpdates function.